src/IECoreHoudini/SOP_SceneCacheSource.cpp

```diff
@@ -311,8 +311,8 @@ void SOP_SceneCacheSource::loadObjects( const IECore::SceneInterface *scene, Ima
 	{
 		return;
 	}
-	
-	if ( scene->hasObject() && UT_String( scene->name() ).multiMatch( params.shapeFilter ) && tagged( scene, params.tagFilter ) )
+
+	if ( scene->hasObject() && UT_String( getFullScenePath( scene ) ).multiMatch( params.shapeFilter ) && tagged( scene, params.tagFilter ) )
```

Comment:
The syntax also allows matching by ^ , which will remove results from the previous match.
`/tree/branchA/twigAA
/tree/branchA/twigAB`

`^twigAA` would match everything, if we added the results of checking the full path and the short name. If we checked the short path first, `^twigAA` would match `/tree/branchA/twigAB` correctly, but then `^/tree/branchA/twigAA` would match `twigAA`.

There could also be some ambiguity.

`/body/leg/smalllegarmour
/body/leg/largelegarmour
/body/leg/foot`

Should `*leg*` also match `/body/leg/foot`, or should it match only `largelegarmour` and `smalllegarmour`.

I'll add the concatenation the names to the recursion.